service/src/test/java/bio/terra/pipelines/stairway/imputation/PrepareImputationInputsStepTest.java

```diff
@@ -0,0 +1,98 @@
+import static org.mockito.Mockito.when;
+
+import bio.terra.pipelines.service.ImputationService;
+import bio.terra.pipelines.testutils.BaseEmbeddedDbTest;
+import bio.terra.pipelines.testutils.StairwayTestUtils;
+import bio.terra.pipelines.testutils.TestUtils;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepStatus;
+import com.fasterxml.jackson.core.type.TypeReference;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class PrepareImputationInputsStepTest extends BaseEmbeddedDbTest {
+
+  @Autowired private ImputationService imputationService;
+  @Mock private FlightContext flightContext;
+
+  private final UUID testJobId = TestUtils.TEST_NEW_UUID;
+
+  private SimpleMeterRegistry meterRegistry;
+
+  @BeforeEach
+  void setup() {
+    var inputParameters = new FlightMap();
+    var workingMap = new FlightMap();
+
+    when(flightContext.getInputParameters()).thenReturn(inputParameters);
+    when(flightContext.getWorkingMap()).thenReturn(workingMap);
+
+    meterRegistry = new SimpleMeterRegistry();
+    Metrics.globalRegistry.add(meterRegistry);
+  }
+
+  @AfterEach
+  void tearDown() {
+    meterRegistry.clear();
+    Metrics.globalRegistry.clear();
+  }
+
+  @Test
+  void doStepSuccess() throws InterruptedException {
+    // setup
+    when(flightContext.getFlightId()).thenReturn(testJobId.toString());
+
+    StairwayTestUtils.constructCreateJobInputs(flightContext.getInputParameters());
+
+    // make sure the full inputs are not populated before the step is executed
+    assertNull(
+        flightContext
+            .getWorkingMap()
+            .get(RunImputationJobFlightMapKeys.ALL_PIPELINE_INPUTS, new TypeReference<>() {}));
+
+    // do the step
+    var prepareImputationInputsStep = new PrepareImputationInputsStep(imputationService);
+    var result = prepareImputationInputsStep.doStep(flightContext);
+
+    // get info from the flight context to run checks
+    FlightMap inputParams = flightContext.getInputParameters();
+    FlightMap workingMap = flightContext.getWorkingMap();
+
+    assertEquals(StepStatus.STEP_RESULT_SUCCESS, result.getStepStatus());
+
+    // make sure the full map of inputs was prepared
+    Map<String, Object> userProvidedInputs =
+        inputParams.get(
+            RunImputationJobFlightMapKeys.USER_PROVIDED_PIPELINE_INPUTS, new TypeReference<>() {});
+    Map<String, Object> fullInputs =
+        workingMap.get(RunImputationJobFlightMapKeys.ALL_PIPELINE_INPUTS, new TypeReference<>() {});
+    assertNotNull(fullInputs);
+    assertNotNull(userProvidedInputs);
+    assertTrue(fullInputs.size() > userProvidedInputs.size());
```

Comment:
should we do any more tests here to be confident the full inputs are being created correctly?